### PR TITLE
feat: [IE-106]: Adding manifest yamls for respective services

### PR DESCRIPTION
**Pull request by Vbhatia20**
## Describe your changes
This change enables single pipelines to be used for building independent services. Each manifest yamls has metadata associated with respective service to enable the build process to go through.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47991)

### CODE REVIEW

**Comment by Vbhatia20** [comment]
trigger uts

**Comment by Vbhatia20** [comment]
trigger messagecheck

**Comment by Vbhatia20** [comment]
trigger messagecheck

**Comment by Vbhatia20** [comment]
trigger messagecheck

**Review by arya-harness — APPROVED** [review]

**Comment by Vbhatia20** [comment]
trigger commonchecks

**Comment by Vbhatia20** [comment]
trigger commonchecks

**Review by arya-harness — APPROVED** [review]

**Review by arya-harness — APPROVED** [review]
> Reviewed 9 of 14 files at r1, 3 of 4 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47991)* status: Waiting on team(s): sto,del-team.

**Comment by Vbhatia20** [comment]
trigger uts

File: 310-iacm-manager/config/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+BUILD_SCRIPT_PATH: 310-iacm-manager/build/feature_build.sh
+DOCKER_FILE_PATH: /harness/harness-core/dist/310-iacm-manager/Dockerfile-ce-nextgen-cie-jdk
+CONTEXT_PATH: /harness/harness-core/dist/310-iacm-manager
+ARTIFACT_REPO: harness/feature-cenextgen-service-openjdk-8u242
+HELM_CHART_REPO: harness-core/310-iacm-manager/chart
+HELM_PUBLISH_REPO: us-west1-docker.pkg.dev/platform-205701/harness-helm-artifacts
+BUILD_STEP_RESOURCE_CPU: 5000m
+BUILD_STEP_RESOURCE_MEMORY: 15Gi
```
Comment (hitesharinga) [inline 310-iacm-manager/config/manifest.yaml:8]: This looks bit on higher side.

File: 315-sto-manager/config/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+BUILD_SCRIPT_PATH: 315-sto-manager/build/feature_build.sh
+DOCKER_FILE_PATH: /harness/dist/sto-manager/Dockerfile-gcr-ubi
+CONTEXT_PATH: /harness/dist/sto-manager
+ARTIFACT_REPO: harness/saas-openjdk-8u242/sto-manager
+HELM_CHART_SOURCE_DIRECTORY: sto-manager/chart
+HELM_REPO_NAME: us-west1-docker.pkg.dev/platform-205701/harness-helm-artifacts
+BUILD_STEP_RESOURCE_CPU: 5000m
+BUILD_STEP_RESOURCE_MEMORY: 15Gi
```
Comment (hitesharinga) [inline 315-sto-manager/config/manifest.yaml:8]: Same, Resource looks bit on higher side.

File: pipeline-service/config/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+BUILD_SCRIPT_PATH: pipeline-service/build/feature_build.sh
+DOCKER_FILE_PATH: /harness/harness-core/dist/pipeline-service/Dockerfile-cie-jdk
+CONTEXT_PATH: /harness/harness-core/dist/pipeline-service
+ARTIFACT_REPO: harness/feature-pipeline-service-openjdk-8u242
+HELM_CHART_SOURCE_DIRECTORY: harness-core/pipeline-service/chart
+HELM_REPO_NAME: us-west1-docker.pkg.dev/platform-205701/harness-helm-artifacts
+BUILD_STEP_RESOURCE_CPU: 5000m
+BUILD_STEP_RESOURCE_MEMORY: 15Gi
```
Comment (hitesharinga) [inline pipeline-service/config/manifest.yaml:8]: Resources

File: platform-service/config/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+BUILD_SCRIPT_PATH: platform-service/build/feature_build.sh
+DOCKER_FILE_PATH: /harness/harness-core/dist/platform-service/Dockerfile-cie-jdk
+CONTEXT_PATH: /harness/harness-core/dist/platform-service
+ARTIFACT_REPO: harness/feature-platform-service-openjdk-8u242
+HELM_CHART_SOURCE_DIRECTORY: harness-core/platform-service/chart
+HELM_REPO_NAME: us-west1-docker.pkg.dev/platform-205701/harness-helm-artifacts
+BUILD_STEP_RESOURCE_CPU: 5000m
+BUILD_STEP_RESOURCE_MEMORY: 15Gi
```
Comment (hitesharinga) [inline platform-service/config/manifest.yaml:8]: Resources

**Review by hitesharinga — DISMISSED** [review]

**Comment by Vbhatia20** [comment]
trigger uts

**Review by mbatchelor — APPROVED** [review]

**Pull request merged**